google-cloud-firestore/lib/google/cloud/firestore/watch/listener.rb

```diff
@@ -288,7 +291,8 @@ def background_run
           rescue RestartStream
             retry
           rescue StandardError => e
-            raise Google::Cloud::Error.from_error(e)
+            @parent.error! e
+            retry
```

Comment:
Do we need to guard against infinite retry? This is kind of a catch-all error, so what would happen if, for example, something goes chronically wrong when calling `listen` on the client?

Reply:
Hmm, I think I made a mistake here, bringing `retry` in from a source impl. Should the method return instead?

Reply:
Or should it raise the error after reporting it?

Reply:
I'm restoring the handling to raise the error (rather than retry), but adding the reporting of the error before the raise.